tools/bumper/parser.go

```diff
@@ -0,0 +1,114 @@
+	Metadata 	metadata
+}
+
+type ComponentsList struct {
+	Multus component `yaml:"multus"`
+	LinuxBridge component `yaml:"linux-bridge"`
+	BridgeMarker component `yaml:"bridge-marker"`
+	Kubemacpool component `yaml:"kubemacpool"`
+	Nmstate component `yaml:"nmstate"`
+	OvsCni component `yaml:"ovs-cni"`
+	MacvtapCni component `yaml:"macvtap-cni"`
+}
+
+type ComponentsConfig struct {
+	Components ComponentsList
+}
+
+func parseComponentsYaml(componentsConfigPath string) (map[string]component, error) {
+	config := ComponentsConfig{}
+
+	Logger.Printf("Opening components config %s", componentsConfigPath)
+	componentsData, err := ioutil.ReadFile(componentsConfigPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to open file %s", componentsConfigPath)
+	}
+
+	Logger.Printf("Unmarshaling components config %s", componentsConfigPath)
+	err = yaml.Unmarshal(componentsData, &config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to Unmarshal %s", componentsConfigPath)
+	}
+
+	if config == (ComponentsConfig{}) {
+		return nil, fmt.Errorf("Failed to Unmarshal %s. Output is empty", componentsConfigPath)
+	}
+
+	return convertToComponentsMap(config.Components), nil
+}
+
+func convertToComponentsMap(components ComponentsList) map[string]component {
+	v := reflect.ValueOf(components)
+
+	componentsMap := make(map[string]component)
+
+	for i := 0; i < v.NumField(); i++ {
+		componentsMap[v.Type().Field(i).Name] = v.Field(i).Interface().(component)
+	}
+
+	return componentsMap
+}
+
+func convertToComponentsConfig(componentsMap map[string]component) *ComponentsConfig {
+	var config = new(ComponentsConfig)
+	config.Components.Multus = componentsMap["Multus"]
+	config.Components.LinuxBridge = componentsMap["LinuxBridge"]
+	config.Components.BridgeMarker = componentsMap["BridgeMarker"]
+	config.Components.Kubemacpool = componentsMap["Kubemacpool"]
+	config.Components.Nmstate = componentsMap["Nmstate"]
+	config.Components.OvsCni = componentsMap["OvsCni"]
+	config.Components.MacvtapCni = componentsMap["MacvtapCni"]
+
+	return config
+}
+
+func updateComponentsYaml(componentsConfigPath string, componentsMap map[string]component) error {
+	config := convertToComponentsConfig(componentsMap)
+
+	Logger.Printf("Marshaling components config")
+	yamlConfig, err := yaml.Marshal(config)
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal updated components config")
+	}
+
+	err = ioutil.WriteFile(componentsConfigPath, yamlConfig, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to write to file %s", componentsConfigPath)
+	}
+
+	return nil
+}
+
+func printCurrentComponentParams(component component) error {
+	componentJson, err := json.Marshal(component)
+	if err != nil {
+		return errors.Wrap(err, "Failed to print component params")
+	}
+
+	var prettyJSON bytes.Buffer
+	err = json.Indent(&prettyJSON, componentJson, "", "\t")
+	Logger.Printf("Current component config:\n%s", string(prettyJSON.Bytes()))
+	return nil
```

Comment:
You can use directly https://golang.org/pkg/encoding/json/#MarshalIndent it do Marshall + Indent

Reply:
Done